gh_index_issues_to_dts_portal.py

```diff
@@ -78,30 +93,65 @@ def build_payload(project_records, project_issues):
     for issue in project_issues:  # iterate over gh issues
         pipeline = find_pipeline_by_issue(zenhub_metadata, issue.number)
 
+        last_comment_body = None
+        last_comment_date = None
+        comments = issue.get_comments()
+        # an issue often has more than one assignee, this returns the list of users assigned to the issue
+        assignees = issue.assignees
+        assignees_logins = [user.login for user in assignees]
+        assignees_string = " ".join(assignees_logins)
+        comments_list = [comment for comment in comments]
+        if len(comments_list) > 0:
+            last_comment = comments_list[-1]
+            last_comment_body = last_comment.body
+            last_comment_date = last_comment.created_at
+
         # ZH metadata does not include closed issues
         if issue.state == "closed":
             pipeline = "Closed"
 
         knack_record = find_knack_record_by_issue(project_records, issue.number)
 
         if knack_record:
+            update_record = False
             issue_payload = {"id": knack_record["id"]}
             title_knack = knack_record[KNACK_TITLE_FIELD]
             pipeline_knack = knack_record[KNACK_PIPELINE_FIELD]
+            last_comment_date_knack = knack_record[KNACK_COMMENT_DATE_FIELD]
+            assignee_knack = (
+                knack_record[KNACK_ISSUE_ASSIGNEE]
+                if knack_record[KNACK_ISSUE_ASSIGNEE]
+                else ""
+            )
 
             if title_knack != issue.title:
                 issue_payload[KNACK_TITLE_FIELD] = issue.title
+                update_record = True
             if pipeline_knack != pipeline:
                 issue_payload[KNACK_PIPELINE_FIELD] = pipeline
-            if title_knack != issue.title or pipeline_knack != pipeline:
+                update_record = True
+            if are_timestamps_different(last_comment_date_knack, last_comment_date):
```

Comment:
I found myself in a situation where I edited a **Most recent comment** in `object_30` but this code was not seeing that the **Comment Date** (`field_689`) was different because I didn't edit that field. Once I edited the date field in the Knack table to be one day in the past, I was able to restore the comment date and most recent comment that is fetch from GitHub.

I'm not sure if I'm thinking about this right, but what would happen if two comments were added on the same day in the GitHub issue? I think the code might not see the update.

Reply:
hm so the dag runs once a day and I was thinking it would pick up the most recent comment then, since it wouldnt have seen the first comment yet. But there could be an edge case (like how John has seen about timezones too) where it picks up a comment but then the dag runs and doesn't pick up the newest...I think I need to write the times out to see

Reply:
perhaps we don't need to compare timestamps, we can just check if the comment is different? that's all that matters, right?